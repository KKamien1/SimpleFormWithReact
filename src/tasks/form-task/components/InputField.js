
import React from 'react';
import ErrorMessage from './ErrorMessage'

export default function InputField({ name, value, required, children, onChange, onBlur, touched, errors }) {
    return (
        <div className="form__input-field">
            <label>{children}
                <input
                    name={name}
                    value={value}
                    onChange={onChange}
                    onBlur={onBlur}
                    type='text'
                    required={required}
                />
                {touched && <ErrorMessage errors={errors} />}
            </label>
        </div>
    )
}
