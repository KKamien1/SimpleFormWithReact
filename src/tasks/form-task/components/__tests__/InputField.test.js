import React from 'react';
import { shallow } from 'enzyme';
import { render, screen } from '@testing-library/react';

import InputField from '../ErrorMessage'

describe('InputField component', () => {
    test('should match to snapshot', () => {
        const component = shallow(
            <InputField
                errors={['Test message']}
            />);
        expect(component.html()).toMatchSnapshot();
    })


})
