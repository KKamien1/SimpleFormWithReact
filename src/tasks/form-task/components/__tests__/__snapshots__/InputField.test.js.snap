// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField component should match to snapshot 1`] = `"<span class=\\"form__error-message\\">Test message</span>"`;
